app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -825,6 +828,17 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         // and let it try to load whatever was entered.
         if ((!forceSearch && searchTermOrURL.isUrl()) || engine == null) {
             loadUrlUseCase.invoke(searchTermOrURL.toNormalizedUrl(), flags)
+
+            if (requestDesktopMode == true) {
+                val requestDesktopSiteUseCase =
+                    components.useCases.sessionUseCases.requestDesktopSite
+                requestDesktopSiteUseCase.invoke(requestDesktopMode)
+                components.core.store.dispatch(
+                    ContentAction.UpdateDesktopModeAction(
+                        components.core.store.state.selectedTabId.toString(), true
+                ))
+                settings().openNextTabInDesktopMode = false
```

Comment:
is this resetting the mode? it's a bit confusing having this within the `if (requestDesktopMode == true)` block

Reply:
Yes. I added a short comment.